project-metrics/metrics_service/server.py

```diff
@@ -4,27 +4,46 @@
 Used to launch the REST and Cron servers.
 """
 
+import logging
+import os
 import flask
 from flask_api import status
-import os
 
 from metrics import base
+from database import commit_scraper
 
+logging.getLogger().setLevel(logging.INFO)
 app = flask.Flask(__name__)
 
 
+@app.route('/_cron/update_data/commits')
+def scrape_latest_commits():
+  # This header is added to cron requests by GAE, and stripped from any external
+  # requests. See https://cloud.google.com/appengine/docs/standard/python3/scheduling-jobs-with-cron-yaml#validating_cron_requests
+  if not flask.request.headers.get('X-Appengine-Cron'):
+    return 'Attempted to access internal endpoint.', status.HTTP_403_FORBIDDEN
+  commit_scraper.CommitScraper().scrape_since_latest()
+  return 'Successfully scraped latest commits.', status.HTTP_200_OK
+
+
 @app.route('/_cron/recompute/<metric_cls_name>')
 def recompute(metric_cls_name):
   # This header is added to cron requests by GAE, and stripped from any external
   # requests. See https://cloud.google.com/appengine/docs/standard/python3/scheduling-jobs-with-cron-yaml#validating_cron_requests
   if not flask.request.headers.get('X-Appengine-Cron'):
     return 'Attempted to access internal endpoint.', status.HTTP_403_FORBIDDEN
   try:
-    metric_cls = base.Metric.active_metrics[metric_cls_name]
+    metric_cls = base.Metric.get_metric(metric_cls_name)
   except KeyError:
+    logging.error('No active metric found for %s.', metric_cls_name)
     return ('No active metric found for %s.' % metric_cls_name,
             status.HTTP_404_NOT_FOUND)
-  metric_cls().recompute()
+  logging.info('Recomputing %s.', metric_cls_name)
+  try:
+    metric_cls().recompute()
+  except Exception as error:
```

Comment:
Can you limit this? Catching `Exception` means also catching `SystemExit`, for example...

Reply:
The SystemExit exception inherits from BaseException instead of StandardError or Exception so that it is not accidentally caught by code that catches Exception. This try/except allows the cron to catch any type of error, log it appropriately, and return the proper response/status.